src/matrixrtc/NewMembershipManager.ts

```diff
@@ -0,0 +1,859 @@
+/*
+Copyright 2025 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { EventType } from "../@types/event.ts";
+import { UpdateDelayedEventAction } from "../@types/requests.ts";
+import type { MatrixClient } from "../client.ts";
+import { UnsupportedEndpointError } from "../errors.ts";
+import { HTTPError, MatrixError } from "../http-api/errors.ts";
+import { logger as rootLogger } from "../logger.ts";
+import { type Room } from "../models/room.ts";
+import { sleep } from "../utils.ts";
+import { type CallMembership, DEFAULT_EXPIRE_DURATION, type SessionMembershipData } from "./CallMembership.ts";
+import { type Focus } from "./focus.ts";
+import { isLivekitFocusActive } from "./LivekitFocus.ts";
+import { type MembershipConfig } from "./MatrixRTCSession.ts";
+
+const logger = rootLogger.getChild("MatrixRTCSessionManager");
```

Comment:
My bad, I realise that this is part of MatrixRTCSession not MatrixRTCSessionManager:

```suggestion
const logger = rootLogger.getChild("MatrixRTCSession");
```